plugins/inputs/p4runtime/p4runtime.go

```diff
@@ -0,0 +1,240 @@
+package p4runtime
+
+import (
+	"context"
+	"crypto/tls"
+	_ "embed"
+	"fmt"
+	"io"
+	"strconv"
+	"sync"
+
+	p4ConfigV1 "github.com/p4lang/p4runtime/go/p4/config/v1"
+	p4v1 "github.com/p4lang/p4runtime/go/p4/v1"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/influxdata/telegraf"
+	internaltls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const (
+	defaultDeviceID = "1"
+	defaultAddress  = "127.0.0.1:9559"
+)
+
+type P4runtime struct {
+	Address      string          `toml:"endpoint"`
+	DeviceID     string          `toml:"device_id"`
+	CounterNames []string        `toml:"counter_names"`
+	Log          telegraf.Logger `toml:"-"`
+	EnableTLS    bool            `toml:"enable_tls"`
+	internaltls.ClientConfig
+
+	conn           *grpc.ClientConn
+	client         p4v1.P4RuntimeClient
+	deviceIDParsed uint64
+	wg             sync.WaitGroup
+}
+
+func (*P4runtime) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *P4runtime) Start(telegraf.Accumulator) error {
```

Comment:
Let's take all of the checking of device ID and Address and put that in init please. That should only happen the very first time the plugin starts.

Additionally, let's avoid the need for the `deviceIDParsed` variable. Store DeviceID as an uint64 from the start.

The Start function can then be return p.newP4RuntimeClient()

Reply:
Removed `deviceIDParsed` variable, remainings moved  to `Init()`